src/dmd/parse.d

```diff
@@ -5081,6 +5081,16 @@ final class Parser(AST) : Lexer
     L1:
         switch (token.value)
         {
+        case TOK.goesTo:
+            if (requireDo)
+                error("missing `do { ... }` after `in` or `out`");
+            const returnloc = token.loc;
+            nextToken();
+            f.fbody = new AST.ReturnStatement(returnloc, parseExpression());
+            check(TOK.semicolon);
+            f.endloc = endloc;
```

Comment:
I think I did this wrong. I just copied from below but p sure this is specifically for }.... should I instead set it to `token.loc` for the semicolon?

Reply:
Seems reasonable to me, and I can't think of any more appropriate token to point it at. It seems to be used only for line mappings for debug info (and I supposed DMD as a Library for whatever).

You should probably update [this comment](https://github.com/dlang/dmd/blob/1339fb58c007ac20eaaeff9549607f9b593a98a6/src/dmd/parse.d#L288) too.

Reply:
That comment is the one on the parser class, this is on the f object which is the FuncDeclaration class.  (It was that comment that tipped me I was doing the wrong thing though.)